src/client/state/navigation.js

```diff
@@ -212,13 +199,6 @@ class Navigation extends EventEmitter {
     if (categorizedSpaces.has(spaceId)) {
       const categories = roomList.getCategorizedSpaces([spaceId]);
 
-      const latestSelectedRoom = this._getLatestSelectedRoomId([...categories.keys()]);
-
-      if (latestSelectedRoom) {
-        this._selectRoom(latestSelectedRoom);
-        return;
-      }
-
```

Comment:
This code get the last selected room when a selected space is marked as categorized. It takes all those spaces as categories and finds room which was most recently selected.